src/main/java/org/jenkinsci/plugins/workinghours/model/TimeRange.java

```diff
@@ -23,143 +23,76 @@
  */
 package org.jenkinsci.plugins.workinghours.model;
 
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-import hudson.Extension;
-import hudson.model.AbstractDescribableImpl;
-import hudson.model.Descriptor;
-import hudson.util.FormValidation;
-import hudson.util.ListBoxModel;
-import java.text.DateFormatSymbols;
-import java.time.LocalTime;
 import java.util.Calendar;
-import org.kohsuke.stapler.DataBoundConstructor;
-import org.kohsuke.stapler.QueryParameter;
+
+import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.workinghours.ValidationResult;
 
 /**
  * Encapsulates a time range, which matches a times on a particular day of the
  * week that occur between a start and end time.
  *
  * @author jxpearce@godaddy.com
  */
-public class TimeRange extends AbstractDescribableImpl<TimeRange> {
+public class TimeRange {
+
+    private static final String FIELD_START_TIME = "startTime";
+    private static final String FIELD_END_TIME = "endTime";
+    private static final String FIELD_DAY_OF_WEEK = "dayOfWeek";
+
+    public static ValidationResult validateTimeRange(JSONObject targetJson) {
+        if (!(targetJson.containsKey(FIELD_START_TIME))) {
+            return new ValidationResult(false, FIELD_START_TIME, "is needed");
+        } else if (!(targetJson.get(FIELD_START_TIME) instanceof Number)) {
+            return new ValidationResult(false, FIELD_START_TIME, "is not a number");
+        } else if (targetJson.getInt(FIELD_START_TIME) > 2400 || targetJson.getInt(FIELD_START_TIME) < 0) {
+            return new ValidationResult(false, FIELD_START_TIME, "should be between 0 and 2400");
+        }
 
-    private TimeRangeDataContainer dataContainer = null;
-    private String jsonData = "";
-    private static Gson gson = null;
+        if (!(targetJson.containsKey(FIELD_END_TIME))) {
+            return new ValidationResult(false, FIELD_END_TIME, "is needed");
+        } else if (!(targetJson.get(FIELD_END_TIME) instanceof Number)) {
+            return new ValidationResult(false, FIELD_END_TIME, "is not a number");
+        } else if (targetJson.getInt(FIELD_END_TIME) > 2400 || targetJson.getInt(FIELD_END_TIME) < 0) {
+            return new ValidationResult(false, FIELD_END_TIME, "should be between 0 and 2400");
+        } else if (targetJson.getInt(FIELD_END_TIME) < targetJson.getInt(FIELD_START_TIME)) {
+            return new ValidationResult(false, FIELD_END_TIME, "should be after start time");
+        }
 
-    /**
-     * Get the actual time range data.
-     * @return TimeRangeDataContainer
-     */
-    public TimeRangeDataContainer getDataContainer() {
-        return dataContainer;
-    }
+        if (!(targetJson.containsKey(FIELD_DAY_OF_WEEK))) {
+            return new ValidationResult(false, FIELD_DAY_OF_WEEK, "is needed");
```

Comment:
These strings need to be localizable in case this plugin is ever translated into another language